lib/fizzy/execute.cpp

```diff
@@ -535,34 +536,36 @@ inline bool invoke_function(const FuncType& func_type, uint32_t func_idx, Instan
     return true;
 }
 
-}  // namespace
-
-ExecutionResult execute(
+template <bool MeteringEnabled>
+ExecutionResult execute_local_function(
```

Comment:
Maybe consider specialising `invoke` as well so that we generate one less branch?

Reply:
Which branch? I think `invoke_function` does not change with `MeteringEnabled`... It still needs `ExecutionContext` to count depth.